chainer/links/normalization/batch_normalization.py

```diff
@@ -62,32 +67,41 @@ class BatchNormalization(link.Link):
 
     """
 
-    def __init__(self, size, decay=0.9, eps=2e-5, dtype=numpy.float32,
-                 use_gamma=True, use_beta=True,
-                 initial_gamma=None, initial_beta=None):
+    def __init__(self, size=None, decay=0.9, eps=2e-5,
+                 dtype=numpy.float32, use_gamma=True, use_beta=True,
+                 initial_gamma=None, initial_beta=None, axis=1):
         super(BatchNormalization, self).__init__()
-        self.avg_mean = numpy.zeros(size, dtype=dtype)
```

Comment:
It changes the behavior of the documented API `self.avg_mean` slightly because after this change, we cannot get this attribute immediately after instantiation when `size` is not `None`. So, please call `self._initialize_param` in `__init__` when  `size` is not `None`.